homeassistant/components/ambiclimate/config_flow.py

```diff
@@ -124,7 +124,7 @@ def _generate_oauth(self):
         )
 
     def _cb_url(self):
-        return f"{get_url(self.hass)}{AUTH_CALLBACK_PATH}"
+        return f"{get_url(self.hass, prefer_external=True)}{AUTH_CALLBACK_PATH}"
```

Comment:
We shouldn't do this. We should try to return the same URL as the user is currently on to visit HA like this https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/config_entry_oauth2_flow.py#L130-L138

Reply:
The mock fixture returns `10.10.10.10` for the source ip. 

All the tests were expecting `assert result["data"]["callback_url"] == "https://example.com/api/ambiclimate"` which looks like the external url, but it was not asking for external url.

AFAICT by digging through the git history it looks like it was created before there was an option to ask which one the integrations prefers and was never updated to ask for it.

Reply:
ok